relay-server/src/actors/outcome.rs

```diff
@@ -475,24 +479,40 @@ pub enum OutcomeError {
 struct HttpOutcomeProducer {
     config: Arc<Config>,
     unsent_outcomes: Vec<TrackRawOutcome>,
-    pending_flush_handle: Option<SpawnHandle>,
+    // Sleep must be `Box<Pin>` in order for it to be Unpin.
+    // See: https://docs.rs/tokio/1.21.0/tokio/time/struct.Sleep.html#examples
+    pending_flush_handle: Either<future::Pending<()>, Pin<Box<tokio::time::Sleep>>>,
 }
 
 impl HttpOutcomeProducer {
     pub fn create(config: Arc<Config>) -> Result<Self, ServerError> {
         Ok(Self {
             config,
             unsent_outcomes: Vec::new(),
-            pending_flush_handle: None,
+            pending_flush_handle: Either::Left(future::pending()),
         })
     }
+
+    pub fn start(mut self) -> ServiceAddr<Self> {
+        let (tx, mut rx) = mpsc::unbounded_channel::<HttpOutcomeProducerMessages>();
+
+        tokio::spawn(async move {
+            loop {
+                tokio::select! {
+                    Some(message) = rx.recv() => self.handle_message(message),
+                    () = &mut self.pending_flush_handle => self.send_batch(),
```

Comment:
This is probably correct but does make me a bit nervous that this branch could be accidentally disabled.  I'd have written it as `_ = &mut ...`

Reply:
Can you elaborate on what you mean by "accidentally disabling this branch"? 

We originally had this as `_ = &mut ...`, but then aligned it with the [official examples](https://docs.rs/tokio/latest/tokio/time/struct.Sleep.html#examples).

Reply:
I'd be worried if this somehow evaluates to something else you'd accidentally disable the branch.  But I guess the typechecker would catch that anyway, and the outcome of the sleep future is well defined.  so fine, ignore me